pkg/controller/provider/container/ovirt/client.go

```diff
@@ -219,3 +219,156 @@ func GetInsecureSkipVerifyFlag(secret *core.Secret) bool {
 
 	return insecureSkipVerify
 }
+
+func TestDownloadOvfStore(secret *core.Secret, log logr.Logger) (ok bool, err error) {
+
+	// Create the connection to the server:
+	conn, err := ovirtsdk.NewConnectionBuilder().
+		URL(string(secret.Data["url"])).
+		Username(string(secret.Data["user"])).
+		Password(string(secret.Data["password"])).
+		Insecure(true).
+		Compress(true).
+		Timeout(time.Second * 30).
+		Build()
+	if err != nil {
+		log.Error(err, "Failed establishing connection to oVirt")
+		err = liberr.Wrap(err)
+		return
+	}
+	defer conn.Close()
+
+	//Find OVF STORE ID
+	disksService := conn.SystemService().DisksService()
+	diskResponse, err := disksService.List().Send()
+	if err != nil {
+		log.Error(err, "Failed to get disks list from oVirt")
+		err = liberr.Wrap(err)
+		return
+	}
+
+	var ovfStoreId string
+	if disks, ok := diskResponse.Disks(); ok {
+		//get OVF STORE ID
+		for _, disk := range disks.Slice() {
+			if diskName, ok := disk.Name(); ok {
+				if diskName == "OVF_STORE" {
+					if diskId, ok := disk.Id(); ok {
+						ovfStoreId = diskId
+					}
+					break
+				}
+			}
+		}
+	}
+	// In case no OVF store was found in the cluster
+	if ovfStoreId == "" {
+		log.Error(err, "OVF Store does not exist on cluster, CA certificate test can't performed")
+		ok = false
```

Comment:
I find this a bit confusing - if we pass this line then `ok` is set to true (line 251) , right?
how about renaming the output argument `ok` to something like `validCA` and then only set it to true in line 369?

Reply:
yes, I agree I've been thinking about that as well, it was supposed to be also an error handler (the way the SDK is set) but in any critical errors we just return err so it's redundant...